addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -0,0 +1,320 @@
+odoo.define('web_editor.ColorPalette', function (require) {
+'use strict';
+
+let core = require('web.core');
+let ColorPickerDialog = require('web.ColorpickerDialog');
+let Dialog = require('web.Dialog');
+let Widget = require('web.Widget');
+let summernoteCustomColors = require('web_editor.rte.summernote_custom_colors')
+
+let qweb = core.qweb;
+let _t = core._t;
+
+let ColorPaletteMixin = {
+    xmlDependencies: ['/web_editor/static/src/xml/snippets.xml'],
+    template: 'web_editor.snippet.option.colorpicker',
+    events: {
+        'click button': '_onColorButtonClick',
+        'mouseenter button': '_onColorButtonEnter',
+        'mouseleave button': '_onColorButtonLeave',
+        'click .o_colorpicker_reset': '_onColorResetButtonClick',
+        'click .o_add_custom_color': '_onCustomColorButtonClick',
+    },
+    /**
+     * @override
+     *
+     * @param {Object} [options]
+     * @param {JQuery} [options.$target] The target element wich has the class or css attribute color on it.
+     * @param {string} [options.colorPrefix='bg-'] Used for the class based colors (theme).
+     * @param {boolean} [options.resetButton=true] Wether to display or not the reset button.
+     * @param {Array<string>} [options.excluded=[]] Sections not to display.
+     */
+    init: function (parent, options) {
+        this.setParent(parent);
+        this._super.apply(this, arguments);
+        this.summernoteCustomColorsArray = [].concat(...summernoteCustomColors).map((color) => this._convertHexToCssRgba(color));
+        this.options = _.defaults(options || {}, {
+            colorPrefix: 'bg-',
+            resetButton: true,
+            excluded: [],
+        });
+    },
+    /**
+     * @override
+     */
+    willStart: async function () {
+        let res = this._super.apply(this, arguments);
+        if (!qweb.has_template('web_editor.colorpicker')) {
+            await this._rpc({
```

Comment:
Here is a reason why your colorpicker always need a parent no ? Or is the rpc duplicated in the editor ?